### PR TITLE
mirrors: refresh the mirrors list

### PR DESCRIPTION
**Pull request by baloo**
###### Motivation for this change

Some mirrors are gone, some migrated to http instead of ftp.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/build-support/fetchurl/mirrors.nix [hunk 1]
```diff
@@ -229,13 +212,6 @@
     "http://archives.fedoraproject.org/pub/archive/fedora/"
   ];
 
-  # Old SUSE distributions.  Unfortunately there is no master site,
-  # since SUSE actually delete their old distributions (see
-  # ftp://ftp.suse.com/pub/suse/discontinued/deleted-20070817/README.txt).
-  oldsuse = [
```
Comment (SuperSandro2000) [inline pkgs/build-support/fetchurl/mirrors.nix:235]: Is this unused in nixpkgs?

File: pkgs/build-support/fetchurl/mirrors.nix [hunk 2]
```diff
@@ -305,13 +279,10 @@
   postgresql = [
     "http://ftp.postgresql.org/pub/"
     "ftp://ftp.postgresql.org/pub/"
-    "ftp://ftp-archives.postgresql.org/pub/"
   ];
 
   metalab = [
-    "ftp://mirrors.kernel.org/metalab/"
     "ftp://ftp.gwdg.de/pub/linux/metalab/"
```
Comment (SuperSandro2000) [inline pkgs/build-support/fetchurl/mirrors.nix:285]: I think we want another mirror here.

File: pkgs/build-support/fetchurl/mirrors.nix (hunk 1 above)
Comment (baloo) [reply pkgs/build-support/fetchurl/mirrors.nix:235]: Does not look like it's used no.

File: pkgs/build-support/fetchurl/mirrors.nix (hunk 2 above)
Comment (baloo) [reply pkgs/build-support/fetchurl/mirrors.nix:285]: finding one was not easy. I still don't know if there is an official mirror list anywhere.

**Pull request merged**